OSDOCS-9882 OCP 4.16 Hosted Control Plane bug fix RNs (#77802)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

* Previously, in disconnected environments, the HyperShift Operator ignored registry overrides. As a consequence, changes to node pools were ignored, and node pools encountered errors. With this update, the metadata inspector works as expected during the HyperShift Operator reconciliation, and the override images are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-34773[*OCPBUGS-34773*])

* Previously, the Hypershift Operator was not using the `RegistryOverrides` mechanism to inspect the image from the internal registry. With this release, the metadata inspector works as expected during the Hypershift Operator reconciliation, and the `OverrideImages` are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-32220[*OCPBUGS-32220*])

* Previously, the OpenShift Cluster Manager container did not have the right TLS Certificates. As a result, image streams could not be used in disconnected deployments. With this update, the TLS Certificates are added as projected volumes. (link:https://issues.redhat.com/browse/OCPBUGS-34390[*OCPBUGS-34390*])

> * Previously, the `azure-kms-provider-active` container in the KAS pod used an entrypoint statement in shell form in the Dockerfile. As a consequence, the container failed. To resolve this issue, use the `exec` form for the entrypoint statement. (link:https://issues.redhat.com/browse/OCPBUGS-33940[*OCPBUGS-33940]*)

Reviewer: ```suggestion
* Previously, the `azure-kms-provider-active` container in the KAS pod used an entrypoint statement in shell form in the Dockerfile. As a consequence, the container failed. To resolve this issue, use the `exec` form for the entrypoint statement. (link:https://issues.redhat.com/browse/OCPBUGS-33940[*OCPBUGS-33940*])
```

* Previously, the inability to find a resource caused re-creation attempts to fail. As a consequence, many `409` response codes were logged in hosted cluster config operator logs. With this update, specific resources were added to the cache so that the hosted cluster config operator does not try to re-create existing resources. (link:https://issues.redhat.com/browse/OCPBUGS-23228[*OCPBUGS-23228*])

* Previously, the pod security violation alert was missing in hosted clusters. With this update, the alert is added to hosted clusters. (link:https://issues.redhat.com/browse/OCPBUGS-31263[*OCPBUGS-31263*])

* Previously, the recycler-pod template in hosted clusters in disconnected environments pointed to `quay.io/openshift/origin-tools:latest`. As a consequence, the recycler pods failed to start. With this update, the recycler pod image now points to the OpenShift Container Platform payload reference. (link:https://issues.redhat.com/browse/OCPBUGS-31398[*OCPBUGS-31398*])

> * With this update, in disconnected deployments, the HyperShift Operator receives the new ICSP/IDMS from the management cluster and adds them to the HyperShift Operator and the Control Plane Operator in every reconciliation loop. The changes to the ICSP/IDMS cause the control-plane-operator pod to be restarted.(link:https://issues.redhat.com/browse/OCPBUGS-29110[*OCPBUGS-29110*])

Reviewer: ```suggestion
* With this update, in disconnected deployments, the HyperShift Operator receives the new ICSP/IDMS from the management cluster and adds them to the HyperShift Operator and the Control Plane Operator in every reconciliation loop. The changes to the ICSP/IDMS cause the control-plane-operator pod to be restarted. (link:https://issues.redhat.com/browse/OCPBUGS-29110[*OCPBUGS-29110*])
```

* Previously, the Hypershift Operator was not using the `RegistryOverrides` mechanism to inspect the image from the internal registry. With this release, the metadata inspector works as expected during the Hypershift Operator reconciliation, and the `OverrideImages` are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-32220[*OCPBUGS-32220*])

* Previously, the OpenShift Cluster Manager container did not have the right TLS Certificates. As a result, image streams could not be used in disconnected deployments. With this update, the TLS Certificates are added as projected volumes. (link:https://issues.redhat.com/browse/OCPBUGS-34390[*OCPBUGS-34390*])

* Previously, the `azure-kms-provider-active` container in the KAS pod used an entrypoint statement in shell form in the Dockerfile. As a consequence, the container failed. To resolve this issue, use the `exec` form for the entrypoint statement. (link:https://issues.redhat.com/browse/OCPBUGS-33940[*OCPBUGS-33940*])

> * Previously, the `konnectivity-agent` daemonset used the `ClusterIP` DNS policy. As a result, when CoreDNS was down, the `konnectivity-agent` pods on the data plane could not resolve the proxy-server-address, and they could fail to `konnectivity-server` in the control plane. With this update, the `konnectivity-agent` daemonset was modified to use `dnsPolicy: Default`. The `konnectivity-agent` uses the host system DNS service to look up the proxy server address, and it does not depend on CoreDNS anymore. (link:https://issues.redhat.com/browse/OCPBUGS-31444[*OCPBUGS-31444*])

Reviewer: ```suggestion
* Previously, the `konnectivity-agent` daemon set used the `ClusterIP` DNS policy. As a result, when CoreDNS was down, the `konnectivity-agent` pods on the data plane could not resolve the proxy-server-address, and they could fail to `konnectivity-server` in the control plane. With this update, the `konnectivity-agent` daemon set was modified to use `dnsPolicy: Default`. The `konnectivity-agent` uses the host system DNS service to look up the proxy server address, and it does not depend on CoreDNS anymore. (link:https://issues.redhat.com/browse/OCPBUGS-31444[*OCPBUGS-31444*])
```

> * Previously, the `konnectivity-agent` daemonset used the `ClusterIP` DNS policy. As a result, when CoreDNS was down, the `konnectivity-agent` pods on the data plane could not resolve the proxy-server-address, and they could fail to `konnectivity-server` in the control plane. With this update, the `konnectivity-agent` daemonset was modified to use `dnsPolicy: Default`. The `konnectivity-agent` uses the host system DNS service to look up the proxy server address, and it does not depend on CoreDNS anymore. (link:https://issues.redhat.com/browse/OCPBUGS-31444[*OCPBUGS-31444*])

Reviewer: I was wondering about the markup for "proxy-server-address", but couldn't find any hits on that in the linked Jira / code fix PR (both mention "proxy-server-url") or QE test (mentions an actual `proxy-server-host` flag), so I wonder if this should more accurately/generically be "proxy server address" or "proxy server URL"? E.g.,:
```suggestion
* Previously, the `konnectivity-agent` daemonset used the `ClusterIP` DNS policy. As a result, when CoreDNS was down, the `konnectivity-agent` pods on the data plane could not resolve the proxy server URL, and they could fail to `konnectivity-server` in the control plane. With this update, the `konnectivity-agent` daemonset was modified to use `dnsPolicy: Default`. The `konnectivity-agent` uses the host system DNS service to look up the proxy server address, and it does not depend on CoreDNS anymore. (link:https://issues.redhat.com/browse/OCPBUGS-31444[*OCPBUGS-31444*])
```

> * Previously, the inability to find a resource caused re-creation attempts to fail. As a consequence, many `409` response codes were logged in hosted cluster config operator logs. With this update, specific resources were added to the cache so that the hosted cluster config operator does not try to re-create existing resources. (link:https://issues.redhat.com/browse/OCPBUGS-23228[*OCPBUGS-23228*])

Reviewer: Seems like this might be an actual Operator name?
```suggestion
* Previously, the inability to find a resource caused re-creation attempts to fail. As a consequence, many `409` response codes were logged in Hosted Cluster Config Operator logs. With this update, specific resources were added to the cache so that the Hosted Cluster Config Operator does not try to re-create existing resources. (link:https://issues.redhat.com/browse/OCPBUGS-23228[*OCPBUGS-23228*])
```

* Previously, the pod security violation alert was missing in hosted clusters. With this update, the alert is added to hosted clusters. (link:https://issues.redhat.com/browse/OCPBUGS-31263[*OCPBUGS-31263*])

> * Previously, the recycler-pod template in hosted clusters in disconnected environments pointed to `quay.io/openshift/origin-tools:latest`. As a consequence, the recycler pods failed to start. With this update, the recycler pod image now points to the OpenShift Container Platform payload reference. (link:https://issues.redhat.com/browse/OCPBUGS-31398[*OCPBUGS-31398*])

Reviewer: ```suggestion
* Previously, the `recycler-pod` template in hosted clusters in disconnected environments pointed to `quay.io/openshift/origin-tools:latest`. As a consequence, the recycler pods failed to start. With this update, the recycler pod image now points to the {produt-title} payload reference. (link:https://issues.redhat.com/browse/OCPBUGS-31398[*OCPBUGS-31398*])
```

> * With this update, in disconnected deployments, the HyperShift Operator receives the new ICSP/IDMS from the management cluster and adds them to the HyperShift Operator and the Control Plane Operator in every reconciliation loop. The changes to the ICSP/IDMS cause the control-plane-operator pod to be restarted. (link:https://issues.redhat.com/browse/OCPBUGS-29110[*OCPBUGS-29110*])

Reviewer: ```suggestion
* With this update, in disconnected deployments, the HyperShift Operator receives the new ICSP/IDMS from the management cluster and adds them to the HyperShift Operator and the Control Plane Operator in every reconciliation loop. The changes to the ICSP/IDMS cause the `control-plane-operator` pod to be restarted. (link:https://issues.redhat.com/browse/OCPBUGS-29110[*OCPBUGS-29110*])
```

* With this update, the `ControllerAvailabilityPolicy` setting becomes immutable after it is set. Changing between `SingleReplica` and `HighAvailability` is not supported. (link:https://issues.redhat.com/browse/OCPBUGS-27282[*OCPBUGS-27282*])

> * With this update, the `machine-config-operator` Custom Resource Definitions (CRDs) are renamed to ensure that resources are being omitted properly in hosted control planes. (link:https://issues.redhat.com/browse/OCPBUGS-34575[*OCPBUGS-34575*])

Reviewer: ```suggestion
* With this update, the `machine-config-operator` custom resource definitions (CRDs) are renamed to ensure that resources are being omitted properly in hosted control planes. (link:https://issues.redhat.com/browse/OCPBUGS-34575[*OCPBUGS-34575*])
```

> * With this update, the size is reduced for audit-log files that are stored in the `kube-apiserver`, `openshift-apiserver` and `oauth-apiserver` pods for hosted control planes. (link:https://issues.redhat.com/browse/OCPBUGS-31106[*OCPBUGS-31106*])

Reviewer: ```suggestion
* With this update, the size is reduced for audit log files that are stored in the `kube-apiserver`, `openshift-apiserver`, and `oauth-apiserver` pods for hosted control planes. (link:https://issues.redhat.com/browse/OCPBUGS-31106[*OCPBUGS-31106*])
```

* Previously, an Amazon Web Services (AWS) policy issue prevented the Cluster API Provider for AWS from retrieving the necessary domain information. As a consequence, installing an AWS hosted cluster with a custom domain failed. With this update, the policy issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-29391[*OCPBUGS-29391*])

* Previously, in disconnected environments, the HyperShift Operator ignored registry overrides. As a consequence, changes to node pools were ignored, and node pools encountered errors. With this update, the metadata inspector works as expected during the HyperShift Operator reconciliation, and the override images are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-34773[*OCPBUGS-34773*])

* Previously, the Hypershift Operator was not using the `RegistryOverrides` mechanism to inspect the image from the internal registry. With this release, the metadata inspector works as expected during the Hypershift Operator reconciliation, and the `OverrideImages` are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-32220[*OCPBUGS-32220*])

> * Previously, the OpenShift Cluster Manager container did not have the correct TLS certificates. As a result, image streams could not be used in disconnected deployments. With this update, the TLS certificates are added as projected volumes. (link:https://issues.redhat.com/browse/OCPBUGS-34390[*OCPBUGS-34390*])

Author: I think so, good catch!
